awx/main/tasks/host_indirect.py

```diff
@@ -0,0 +1,196 @@
+import logging
+from typing import Tuple, Union
+import time
+
+import yaml
+
+import jq
+
+from django.utils.timezone import now, timedelta
+from django.conf import settings
+from django.db import transaction
+
+# Django flags
+from flags.state import flag_enabled
+
+from awx.main.dispatch.publish import task
+from awx.main.dispatch import get_task_queuename
+from awx.main.models.indirect_managed_node_audit import IndirectManagedNodeAudit
+from awx.main.models.event_query import EventQuery
+from awx.main.models import Job
+
+logger = logging.getLogger(__name__)
+
+
+class UnhashableFacts(RuntimeError):
+    pass
+
+
+def get_hashable_form(input_data: Union[dict, list, Tuple, int, float, str, bool]) -> Tuple[Union[Tuple, int, float, str, bool]]:
+    "Given a dictionary of JSON types, return something that can be hashed and is the same data"
+    if isinstance(input_data, (int, float, str, bool)):
+        return input_data  # return scalars as-is
+    if isinstance(input_data, dict):
+        # Can't hash because we got a dict? Make the dict a tuple of tuples.
+        # Can't hash the data in the tuple in the tuple? We'll make tuples out of them too.
+        return tuple(sorted(((get_hashable_form(k), get_hashable_form(v)) for k, v in input_data.items())))
+    elif isinstance(input_data, (list, tuple)):
+        # Nested list data might not be hashable, and lists were never hashable in the first place
+        return tuple(get_hashable_form(item) for item in input_data)
+    raise UnhashableFacts(f'Cannonical facts contains a {type(input_data)} type which can not be hashed.')
+
+
+def build_indirect_host_data(job: Job, job_event_queries: dict[str, str]) -> list[IndirectManagedNodeAudit]:
+    results = {}
+    compiled_jq_expressions = {}  # Cache for compiled jq expressions
+    facts_missing_logged = False
+    unhashable_facts_logged = False
+    for event in job.job_events.filter(task__in=job_event_queries.keys()).iterator():
```

Comment:
@AlanCoding I have tested this with the VMWare collections, but it looks like none of the job events match that query: the task name is either empty or doesn't match the module name.